File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = nums[0],res=sum;

        for(int i=1;i<nums.size();i++){
            if(sum+nums[i]<nums[i]) sum=nums[i];
            else sum+=nums[i];
            res=max(sum,res);
        }
        return res;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res,curr_sum=0,0 
        for val in nums: 
            if curr_sum+val<=val: 
                curr_sum=val 
            else: 
                curr_sum+=val 
            res=max([curr_sum,res])
        return res